### PR TITLE
Segregate the db cache

### PR DESCRIPTION
**Pull request by alxngyn**
fixes issue #161
## Changes in this PR.

- [X] Created a new DB cache file
- [x] Fixed all calls to the cache in DBSize to reference the new DB Cache
- [X] No new functionalities
## Testing this PR.

1. See travis-ci tests
### Expected Output.

```
The command "./run-tests.sh" exited with 0.

Done. Your build exited with 0.
```

@osuosl/devs

### CODE REVIEW

**Comment by Kennric** [comment]
@alxngyn  Looks like this is going to take some heavy rebasing, we changed how all the settings work

File: scheduler.rb [hunk 1]
```diff
@@ -35,12 +36,25 @@
...
     require_relative "plugins/#{p[:name]}/plugin.rb"
-    # For each key in cache
-    cache.keys.each do |key|
-      # Store the node information in the proper table with the plugin's store
-      # method. The plugin object is retrieved from the name string using
-      # Object.const_get. Do not try to store keys that store a datetime object.
-      Object.const_get(p[:name]).new.store key unless key.end_with?('datetime')
+
+    # if DBSIZE plugin, use the db_cache, otherwise use the regular cache
+    if p[:name] == 'DBSize'
```
Comment (Kennric) [inline scheduler.rb]: Making this depend on the name of the plugin seems a little brittle, maybe we should do this by resource? (plugins declare their resource, I believe). Then our caches can correspond to the resource type, a cache for nodes, one for databases, etc.

Comment (Kennric) [inline scheduler.rb]: For reference, something like 
``` 
cache_file = "#{Iam.settings.cache_path}/#{plugin[:resrouce_type]}_cache.json" 
```
Then cache_file will just refer to the file for whatever resource type the plugin is working with

**Comment by Kennric** [comment]
@alxngyn I rebased this with develop, which has some changes to the scheduler for production. Please review my changes and test everything.

**Comment by alxngyn** [comment]
++

**Pull request merged**